chemprop/v2/data/utils.py

```diff
@@ -1,66 +1,173 @@
-from collections import defaultdict
-from typing import Sequence
+import copy
+import itertools
+import logging
+from enum import auto
+from typing import Sequence, Tuple
 
 import numpy as np
-from rdkit.Chem.Scaffolds import MurckoScaffold
-
+from astartes import train_test_split, train_val_test_split
+from astartes.molecules import train_test_split_molecules, train_val_test_split_molecules
 
+from chemprop.v2.cli.utils import LOG_DIR, NOW
 from chemprop.v2.data.datapoints import MoleculeDatapoint
+from chemprop.v2.data.datasets import MoleculeDataset
+from chemprop.v2.data.utils import log_scaffold_stats
+from chemprop.v2.utils.utils import AutoName
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_LOGFILE = f"{LOG_DIR}/{NOW}.log"
+LOG_LEVELS = [logging.ERROR, logging.WARNING, logging.INFO, logging.DEBUG]
+SUBCOMMANDS = []
+
+
+class SplitType(AutoName):
+    CV_NO_VAL = auto()
+    CV = auto()
+    SCAFFOLD_BALANCED = auto()
+    RANDOM_WITH_REPEATED_SMILES = auto()
+    RANDOM = auto()
+    KENNARD_STONE = auto()
+    KMEANS = auto()
 
 
 def split_data(
     data: Sequence[MoleculeDatapoint],
-    split: str = "random",
-    sizes: tuple[float, float, float] = (0.8, 0.1, 0.1),
-    k: int = 5,
-    fold: int = 0,
-):
+    split: str | SplitType,
+    sizes: Tuple[float, float, float] = (0.8, 0.1, 0.1),
+    key_molecule_index: int = 0,
+    seed: int = 0,
+    num_folds: int = 1,
+) -> Tuple[MoleculeDataset, MoleculeDataset, MoleculeDataset]:
+    r"""
+    Splits data into training, validation, and test splits.
+
+    :param data: A sequence of :class:`~chemprop.data.MoleculeDatapoint`.
+    :param split_type: Split type.
```

Comment:
Wondering if we can have the docstring show what split types are available, something like:
`r"""
    Splits data into training, validation, and test splits.
    :param data: A sequence of :class:`~chemprop.data.MoleculeDatapoint`.
    :param split_type: Split type. Choose one of '''
+ str(SplitTypes.keys())
+ r"""
    :param sizes: A length-3 tuple with the proportions of data in the train, validation, and test sets.
    :param key_molecule_index: For data with multiple molecules, this sets which molecule will be considered during splitting.
    :param seed: The random seed to use before shuffling data.
    :param num_folds: Number of folds to create (only needed for "cv" split type).
    :return: A tuple of :class:`~chemprop.data.MoleculeDataset`\ s containing the train,
             validation, and test splits of the data.
    """`

Reply:
This is a cool suggestion but from what I can see on StackOverflow ([here](https://stackoverflow.com/questions/2693883/dynamic-function-docstring)) it's not really possible. I have added reference to the SplitTypes enum in the docstring.